### ISSUE TITLE
"X's favorite" construction

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/docs#650]
In EWT, the current policy is:

- In a noun-headed NP like "my favorite restaurant", the possessive and the adjective "favorite" are sisters under the noun.
- If a possessive and "favorite" are not followed by a noun ("this restaurant is my favorite", "my favorite is the one..."), then "favorite" heads the possessive.

(GUM has 3 instances of the former and no instances of "X's favorite" without a noun.)

I gather the rationale was to follow a canonical NP structure where the noun head is present and promote the adjective where it isn't, but it is not entirely satisfying because this is not a typical case of ellipsis. Syntactically, "favorite" seems freer to have a nounless possessive than most adjectives:

- This is my favorite book.
   * This book is my favorite.
- This is my new book. 
   * *This book is my new.

Semantically, "favorite" expresses a relationship between an opinion-holder and a favored item. "This is my favorite book" doesn't entail "This is my book". (Nor does "This is my favorite green book" entail "This is my book.") The possessive is really acting with respect to "favorite".

Would it be better to make that relationship explicit by having the possessive modify "favorite" even when the noun is present? I.e.

- This is my favorite book: `nmod:poss(favorite, my)`, `amod(book, favorite)`
- This book is my favorite: `nmod:poss(favorite, my)`, `nsubj(favorite, book)`

Some more details:

- "the favorite fruit of founder Steve Jobs": By a parallel argument to the above, it is "favorite" that licenses the possessive of-phrase, so `nmod(fruit, Steve)` could be revised to `nmod(favorite, Steve)`
  * If we're really careful we could also extend this to infinitival modifiers: "my favorite place to grab flowers" is not necessarily my OVERALL favorite of the set of places to grab flowers; it is my favorite with respect to the goal of grabbing flowers. But this might be impractical to implement.
- "my 3rd favorite book" should obviously have "3rd" modifying "favorite". I think it should be the same with "my 3 favorite books", treating the 3 as part of the "favorite" relation (i.e. the top 3 books in the list of favorites).
- "favorite" is tagged as `NOUN` if it has an article and no head noun, e.g. "a favorite of our family". I don't have a strong opinion about this.

(cf. "X's own", #638)

**Comment by Stormur** [comment]
I am actually not so convinced about the annotation change you propose, because I think that _my_ is still modifying _book_, which is the head of the phrase; we can just observe that sometimes _my_ determines a true possession and other times a looser relation with the person it expresses (probably analogous to the alienable/inalienable feature some languages display). In any case, we have a determiner modifying the head _book_. It is dubious if such a semantic criterion should influence the syntactic annotation in this case (I think not).

In any case, it seems to me that the basic UD annotation scheme lacks a proper way to annotate "nesting". In this example the fact you are explaining is probably that _my_ is modifying the phrase _(favourite book)_, and not that the phrase _(my favourite)_ is modifying _book_:

-  `my -> (favourite -> book)` vs. `(my favourite) -> book `

But still, _book_ is the head also _my_ is pointing to. 
(On the other hand, the _favourite_ in _It is my favourite_ seems to behave as a noun, in that you can also say _These are my favourites_, no? So this actually raises the question of determining the POS of _favourite_ and if this construction is of a compound- or apposition-like kind).

But, in general, when we have a sequence of three elements A, B, C, with relations A -> C, B -> C and C head, we don't have an overt representation of the possible actual underlying structures:

- (A B) -> C vs. A -> (B C) vs. plain A -> C, B -> C

This is parallel to what happens with (possibly) nested co-ordinations like A and B and C. Is it:

- (A and B) and C, A and (B and C) or just A and B and C?

In the first and last cases the UD structure will be the same: `conj(A,B)`, `conj(A,C)`.

Another similar case with appositions, from Latin (and no helpful punctuation in the original):

                              Raspertus presbiter lociserbator
                               _Raspertus presbyter warden_

Is it the [presbyter Raspertus] who is also a warden, or a Raspertus who is [presbyter and warden]? In both cases the notation will be `appos(Raspertus,presbiter)`, `appos(Raspertus,lociserbator)`.

So, just to sum it up: I think the basic annotation of UD is OK as it is now in such cases, but that it inherently lacks ways to annotate "nesting". This could however be done somehow by means of enhanced dependencies, for example: `1(favourite,book)`, `2(my,book)`...

**Comment by amir-zeldes** [comment]
I agree with @nschneid that `my favorite` by itself is a different structure, and indeed a noun, since it is pluralizable - we can see that the phrasal head is what take pluralization:

  * My favorite/JJ book/NN
  * My favorite/NN

Because:

  * My favorite/JJ books/NNS
  * My favorites/NNS

I also agree with @Stormur that the issue is one of nesting. Even if semantics were a deciding factor, which I'm not saying, it would still not be 'my favorite' by itself + 'book', since it is not 'a book characterized by being my favorite thing'. As @nschneid put it, it is 'my favorite with respect to...'. So I think the nesting explanation works: it is a 'favorite book' as a phrase, which is relational, and then we specify the required argument, 'favorite_book([of] me)'.

Adding a somewhat convoluted second argument about hierarchy: 'the favorite books of John' does not mean a pluralization of individual things, each of which is his favorite above all others (i.e. compose 'John's favorite' first). It is a set of book things (NP), which together are favored, and then we add John as the argument of favoring the whole set. 

But ultimately I think the argument should be syntactic - this behaves like a normal adjective formally, unless it is nominalized, in which case it becomes a head and exhibits pluralization. In older Germanic this would have been easier to recognize even in the singular case, which looks different when nominalized - so maybe there is also a cross-linguistic argument to only see it as the head when there is no noun.

**Comment by sylvainkahane** [comment]
Thanks @nschneid for these interesting data. They remind me the problem of raising verbs, where we have the same mismatch between semantic and syntactic relations.
If you look at the clause _Peter seems to sleep_, clearly _Peter_ is the syntactic subject of _seems_ (linear position, verb agreement), but it is not its semantic argument: the paraphrase with _It seems that Peter is sleeping_ (as well as other properties such as *_It is to sleep that Peter seems_) shows that the verb TO SEEM has only one semantic argument.
For _my favorite book_, _my_ is a semantic argument of _favorite_: _my favorite book_ is the book which is my favorite (and not necessary my book). But it would be very strange not to analyze _my_ as the determiner of _book_. If we don't, we would have a very uncommon syntactic structure. So I think _my favorite book_ could be an example of raising in the noun phrase.

**Comment by nschneid** [comment]
@Stormur, @amir-zeldes, if I understand correctly you're arguing that the full phrase structure should be [NP my [NP favorite book]], i.e. it is [NP favorite book] that licenses the possessive, but the two levels of nesting of modifiers must get flattened in conversion to dependencies.

I don't necessarily disagree with the phrase structure analysis, but if we think of dependencies as licensing relations, why not say that it is the adjective more than the noun that licenses the possessive in this case?

I guess the answer is that it would be like saying the noun itself doesn't have a determiner, which is not normally permissible for singular count nouns (I bought my (favorite) book, but *I bought book). So yeah, this would be weird.

We could maybe carve out a special exception by treating "favorite" as a determiner and "my" as a predeterminer...but this starts to sound impractical for annotators to remember.

@sylvainkahane: Agreed, the analogy to raising verbs is interesting.

Just for fun, some examples of how "favorite" and superlatives constrain determiners:

- I paid for my/the/a red rose
- I paid for my/?the/?a favorite red rose (possessive preferred; “a” and “the” might be OK depending on context establishing whose favorite, and whether it is the absolute favorite or one of a set)
- I paid for my/the/*a prettiest red rose (superlative normally implies definiteness)

- I paid for 3/some red roses
- I paid for ?3/?some favorite red roses (again, depends on context to infer whose favorite)
- I paid for *3/*some prettiest red roses

- I bought red roses (generic NP)
- *I bought favorite/prettiest red roses

Anyway, I concede that we should stick with the current basic dependencies. But these are cases where bilexical dependencies are weak as a model of grammaticality constraints AND semantics.

As @Stormur and @amir-zeldes, the fact that "favorite" can be pluralized if not followed by a noun is a good argument that it should be tagged as NOUN. It actually is in 7/8 EWT tokens. I can change the other one.

So we're left with the question: Should the __enhanced__ dependencies reflect the raising-like behavior of "favorite", as @sylvainkahane suggests? It seems like a good solution from the perspective of using the enhanced dependencies for semantics.

**Comment by nschneid** [comment]
(https://github.com/UniversalDependencies/UD_English-EWT/commit/d12f83f8fd5cb54cc0cda52eb7b9a30cc49287c9 fixes the 1 token where it should be NOUN rather than ADJ)

**Comment by dan-zeman** [comment]
After reading through this thread I am changing the milestone to "later" because the remaining question is about possible future changes to the guidelines for the enhanced representation.

I wonder whether there is a wider and cross-linguistic formulation of the possible new enhancement. Or is it just one word in one of the 80+ languages (+ possibly the translations of this one word in languages where it behaves similarly)?

**Comment by LarsAhrenberg** [comment]
Just some additional cross-lingual data. In Swedish (and I suspect Norwegian and Danish), the corresponding noun is 'favorit'. When it is used as a nominal head it behaves just as the English 'favorite' but it can only combine with another noun as the first part of a compund:

> _my favorite book --> min favoritbok_

Thus, here it won't even be an independent node of a UD tree. The possessive determiner is definitely the preferred one, but you can without problems use it with an article, although an individual with a 'possessive' relation to the refent can then be inferred from the context

> _En favoritbok/favoritboken är Bibeln (a/the favourite book is the Bible)_

It is somewhat awkward to combine it with other attributes. The translation of _my favorite red rose_ can only be _min röda favoritros_, but that is something I would avoid saying, instead expressing myself in terms such _Of the red roses my favorite is ..._
